src/univariates.jl

```diff
@@ -153,31 +153,28 @@ end
 ##### generic methods (fallback) #####
 
 ## sampling
+# single univariate, no allocation
+rand(rng::AbstractRNG, s::Sampleable{Univariate}) = _rand!(rng, s)
```

Comment:
I still don't quite understand why we need to define all the univariate distributions in terms of `_rand!`: I like to avoid the use of underscore prefix functions where possible. 

Also there is sort of a convention that RNGs are the one thing that don't require a `!` to mutate.

Reply:
That's completely fair, I was just getting lazy (/ for consistency of naming) since the only thing that didn't mutate at this point was single sample univariates as all of the allocation for everything else is now done before you get to the distributions. I'll change it.

Reply:
Thanks. I know how much of a pain these sort of changes are...

Reply:
I'm going ahead and changing it, but I remember my justification now - it was because the RNG was an argument and was being changed...

Reply:
Yeah I know it is a little weird. I can't find the original discussion right now, but it has become the convention for better or worse (see the [the `rand` docs](https://docs.julialang.org/en/v1/stdlib/Random/#Base.rand)). I guess the simplest argument is that anything involving random number will result in mutating the RNG, so it's not a helpful distinction.

Reply:
Absolutely, yes. Anyway, done now.